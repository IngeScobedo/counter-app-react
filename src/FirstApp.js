import PropTypes from 'prop-types';

import React from 'react'

const FirstApp = ( {saludo,subtitle} ) => {
    return (
        <>
          <h1>{saludo}!!!</h1>  
          <p>{subtitle}</p>
        </>
    )
}

FirstApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle: 'Soy un pinche subtitulo'
}

export default FirstApp
