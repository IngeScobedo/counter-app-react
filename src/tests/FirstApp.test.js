import { render } from "@testing-library/react";
import FirstApp from "../FirstApp";
import { shallow } from "enzyme";

describe("Testing - FirstApp Component", () => {
  test("Debe Mostrar un saludo", () => {
    // let saludo = 'que verga quieres';
    // let {getByText} = render( <FirstApp saludo={saludo} /> )
    // expect(getByText(saludo)).toBeInTheDocument();
  });
  test("should render", () => {
    let saludo = "que verga quieres";
    let wrapper = shallow(<FirstApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("should render subtitle correctly", () => {
    let saludo = "que verga quieres";
    let subtitle = "perra madre juan";
    let wrapper = shallow(<FirstApp saludo={saludo} subtitle={subtitle} />);

    let paragraphText = wrapper.find('p').text();
expect(paragraphText).toBe(subtitle);
  });
});
