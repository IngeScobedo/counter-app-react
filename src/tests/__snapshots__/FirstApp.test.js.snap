// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing - FirstApp Component should render 1`] = `
<Fragment>
  <h1>
    que verga quieres
    !!!
  </h1>
  <p>
    Soy un pinche subtitulo
  </p>
</Fragment>
`;
